gitlab/resource_gitlab_service_pipelines_email.go

```diff
@@ -0,0 +1,114 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabServicePipelinesEmail() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabServicePipelinesEmailCreate,
+		Read:   resourceGitlabServicePipelinesEmailRead,
+		Update: resourceGitlabServicePipelinesEmailUpdate,
+		Delete: resourceGitlabServicePipelinesEmailDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceGitlabServicePipelinesEmailImportState,
```

Comment:
I noticed your import and read functions are nearly identical. To save on some code, can use `State: schema.ImportStatePassthrough` here, and then change your read function to get the project from the resource id instead of the `project` attribute, and finally change your `resourceGitlabServicePipelinesEmailSetToState` function to set the `project` attribute. Since the read function is called by Terraform during import, there should be no difference.

Reply:
Well, a bit trickier than I thought. First the ID of the resource and the project ID are different. I could use the project ID though, since you can only have one resource per project

Reply:
I can probably remove the code of the import function. I ended up with that while debugging, but actually I fixed my issue in the go-gitlab package

Reply:
I'm not sure I see where the issue is. (Maybe you already fixed it in a rebase?) It looks like you can use `ImportStatePassthrough`. Here's the relevant section of the docs in case it helps. 😄 
https://www.terraform.io/docs/extend/resources/import.html#importer-state-function

Reply:
The issue is that the read fonction does not have access to the project ID. I use the project ID in omport function for ease of use since services are unique per project and the service ID is pretty hard to retrieve.
So I don't think I can only use the `ImportStatePassthrough` since I need to write the project in the state.

I am not sure though. Does that make sense?

Reply:
That makes me think there is probably a bug in my code